src/hotspot/share/code/relocInfo.hpp

```diff
@@ -818,17 +887,19 @@ class Relocation {
 
 // certain inlines must be deferred until class Relocation is defined:
 
-inline RelocationHolder::RelocationHolder() {
-  // initialize the vtbl, just to keep things type-safe
-  new(*this) Relocation(relocInfo::none);
-}
+inline RelocationHolder::RelocationHolder() :
+  RelocationHolder(Construct(), [&] (void* p) { return ::new (p) Relocation(); })
+{}
```

Comment:
And this would become
```suggestion
inline RelocationHolder::RelocationHolder() : RelocationHolder(Construct<Relocation>())
{}
```

Reply:
Nice suggestion.  That makes the whole pattern a little more palatable, since it puts the magic new in one place instead of N places.